src/librbd/journal/Replay.cc

```diff
@@ -390,6 +390,31 @@ void Replay<I>::handle_event(const journal::AioWriteSameEvent &event,
   }
 }
 
+ template <typename I>
+ void Replay<I>::handle_event(const journal::AioCompareAndWriteEvent &event,
+                              Context *on_ready, Context *on_safe) {
+  CephContext *cct = m_image_ctx.cct;
+  ldout(cct, 20) << ": AIO CompareAndWrite event" << dendl;
+
+  bufferlist cmp_data = event.cmp_data;
+  bufferlist write_data = event.write_data;
+  bool flush_required;
+  auto aio_comp = create_aio_modify_completion(on_ready, on_safe,
```

Comment:
I'd prefer to have ```create_aio_modify_completion``` modified to take a list of filtered error codes where this can pass ```{-EILSEQ}``` and other uses pass ```{}``` so that ```handle_aio_modify_complete``` can translate ```r == <filtered error code>``` back to zero and avoid have the generic handling for ```-EILSEQ``` in ```handle_aio_modify_complete``` and ```Journal::handle_replay_process_safe```